### PR TITLE
HAWQ-333. Remove MD Versioning code

### PR DESCRIPTION
**Pull request by gcaragea**
We will implement a simpler invalidation-based solution instead of versioning. This will be used by the Query Optimizer to manage Metadata Cache.

### CODE REVIEW

**Comment by gcaragea** [comment]
@armenatzoglou @foyzur please take a look.

File: src/backend/gpopt/translate/CTranslatorUtils.cpp [hunk 1]
```diff
@@ -354,21 +353,16 @@ CTranslatorUtils::PmdidWithVersion
 		ullDDLv = 0;
```
Comment (foyzur) [inline src/backend/gpopt/translate/CTranslatorUtils.cpp]: It's already initialized to 0. Do you need this code?

**Comment by foyzur** [comment]
LGTM. Any reason why we cannot revert the previous commits and let Git handle the cleanup job?

**Comment by armenatzoglou** [comment]
LGTM +1

**Comment by gcaragea** [comment]
I removed the extra GUCs and logging / instrumentation code. Please take a look.

**Comment by gcaragea** [comment]
@foyzur The code is well contained, and this is easier than reverting a few tens of commits (in my opinion). Do you have any specific concern?

**Comment by foyzur** [comment]
If it's 10s of commits, then it makes sense.

**Pull request closed without merging**